File: dart/circle.dart
import "dart:math" show pi;

class Circle {
  num x;
  num y;
  num radius;
  String color;
  Circle({this.x: 1, this.y: 1, this.radius: 1, this.color: 'black'});

  num get area => pi * this.radius * this.radius;
  num get circumference => 2 * pi * this.radius;
}

void main() {
  var c = new Circle(x: 1, y: 5);
  print(c.area);
}

File: js/circle.mjs
import assert from "assert"

function Circle(centerX = 0, centerY = 0, radius = 1, color = "black") {
  this.x = centerX
  this.y = centerY
  this.radius = radius
  this.color = color
}

Circle.prototype.area = function () {
  return Math.PI * this.radius * this.radius
}

Circle.prototype.circumference = function () {
  return 2 * Math.PI * this.radius
}

const c = new Circle(1, 5)
assert(c.x === 1 && c.y === 5 && c.radius === 1 && c.color === "black")
assert(c.area() === Math.PI)
assert(c.circumference() === 2 * Math.PI)
assert(Object.getPrototypeOf(c) === Circle.prototype)
assert(c.constructor) === Circle
assert(typeof c === "object")
